apps/nrm/behaviours/set-referral-state.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+module.exports = superclass => class extends superclass {
+  configure(req, res, next) {
+    if (req.sessionModel.get('automatic-referral')) {
+      delete req.form.options.fields['pv-want-to-submit-nrm'].validate;
+    }
+    next();
+  }
+
+  saveValues(req, res, next) {
+    if (req.sessionModel.get('automatic-referral')) {
+      req.form.values['pv-want-to-submit-nrm'] = 'yes';
+    }
+
+    const isReferral = req.form.values['pv-want-to-submit-nrm'] === 'yes';
+
+    req.sessionModel.set('is-referral', isReferral);
```

Comment:
forgive me for ignorance, 
what is the need for `isReferral`? why can't we just use `automatic-referral`?

Reply:
Yes this crossed my mind too. Automatic referral is only when the victim is under 18. isReferral is set for when an application needs to be sent as a referral and not DtN. The reason for having both is to ensure the integrity of the state of the application. 
```
Scenario 1 - Child
Automatic-referral : true
isReferral : true

Scenario 2 - Adult Referral
Automatic-referral : false
isReferral : true

Scenario 3 - Adult DtN
Automatic-referral : false
isReferral : false
```
As you can see they are not the same thing so both are required.

Reply:
Nice one @Alex-Swann